### DISCUSSION TITLE
Focar no Front ou aprofundar no PHP?

### DISCUSSION THREAD

**Discussion by matheusggds** [frontendbr/forum#538] [Discussão]
Bom dia galera,

Tenho uma dúvida de VIDA em relação a qual área investir de verdade e sem medo. Trabalho como dev Front-end a 6 anos, tenho 22, mas de acordo com o tempo fui pesquisando sobre PHP pois para muitos freelas era necessário, e não queria depender de outra pessoa. Hoje em dia onde trabalho eu praticamente só desenvolvendo temas para Wordpress, usando Bootstrap, GULP, css3, html e muitas vezes PHP pois desenvolvo alguns plugins também. A questão é, estou CHEIO de desenvolver praticamente a mesma coisa a 6 anos, claro que com ferramentas novas que fazem nossa performance ser muito melhor, mas a BASE, o Objetivo é quase sempre o mesmo, um site bonito, painel de controle para o usuário, performance boa, rankeamento bom no google. Eu adoro o que faço, mas estou um pouco saturado, e o máximo de salário que consegui atingir foi R$3.000,00 CLT. Sei que não sou um senior front-end, mas pesquiso muito para usar o CSS e HTML da MELHOR FORMA POSSÍVEL, e particularmente acho que faço esta parte muito bem. Mas estou MUITO em dúvida de qual caminho seguir, já estou com 22 anos, formado em análise e desenvolvimento de sistemas, e não sei para onde seguir... Vale a pena investir e investir no FRONT? Ou talvez se aprofundar mais ainda no PHP e ficar BOM nos 2??? Tenho certeza que o mundo do front ainda tem MUITA ferramenta que não conheço e não utilizo... mas a questão é, acho que o limite para um front-end é menor em questão a salários, ferramentas, etc... ou não? Desculpe se não fui claro, estou um pouco confuso sobre o que seguir e focar. Eu particularmente me acho um dev muito bom, tento usar as coisas mais novas possíveis.

**Comment by bernardodiasc** [comment]
A pergunta que voce, e somente voce, deveria procurar resposta é: Por que voce faz o que voce faz?

@matheusggds, com respeito expressarei minha opiniao, tenho visto muitos questionamentos no forum sobre qual caminho seguir dentre tantas opções, o que fazer da vida e tal. Mas não parece util termos tantos posts direcionados para questoes e gostos tao pessoais ou tendencias volúveis do mercado.

22 anos, naturalmente esse questionamento surge, o que fazer da vida, quais os melhores caminhos, etc. Qualquer um que tentar te responder diretamente isso pode mais atrapalhar do que ajudar. Ninguem na historia da humanidade passou ou esta passando pelo mesmo que voce, o que voce vive é unico, as respostas estao ai, e em nenhum outro lugar.

O que voce pode fazer em termos práticos é avaliar tendencias e opçoes, mas investir sem medo vai depender da sua propria analise e coragem. Quem é que vai te dizer que um caminho é melhor que outro pra voce? Como alguem poderia saber?

Ok, questões pragmaticas, o que voce quer fazer nos anos futuros? Ganhar dinheiro? Ser feliz? Entregar qualidade? Não depender de ninguem pra desenvolver soluções inteiras? Não perder tempo com tecnologias que serao em breve extintas? Ou estar à frente dos outros conhecendo linguagens praticamente experimentais? Quer trabalhar pra alguem ou pra si mesmo? Quer ficar de castigo num escritorio ou tirar proveito da internet? Limitar a um conjunto de tecnicas (se ser muito bom naquilo) ou correr o risco de nao ser tao bom a curto prazo estudando varias tecnicas diferentes? Quer trabalhar pra grandes empresas em projetos desafiadores? Zona de conforto lhe parece boa ou má ideia? Por quanto tempo aguenta a pressao?

Sao muitas questoes, cada uma pode te levar pra um caminho revolucionariamente diferente, e ter e ser o melhor de tudo me soa bastante ilusório (nao impossível, mas bem dificil, especialmente se achar que pode te-los a curto prazo). Ao invés de fazermos do forum um divã, podemos avaliar o mercado atual, elaborar perspectivas sobre o futuro da profissão, entender características das tecnologias que fazem umas morrerem e outras sobreviverem ao tempo. Etc.

Se a questão é apenas `front-end vs PHP` ao pé da letra... ao meu ver, nenhuma resposta direta vai ser boa o suficiente, nao é assunto simples. Mas vamos ver quais sao as opiniões sobre ambos os cenários pela experiencia de outros devs. Apesar de que muito da informaçao que voce procura ja foi explorada em varios dos quase 500 posts desse forum, é estudar, avaliar e pesar na sua propria balança.

Boa sorte na sua busca 💪

**Comment by ericdouglas** [comment]
Há um tempo atrás fiz uma análise das vagas no site angel.co e aqui estão alguns destes dados voltados para sua questão: [link](http://docdro.id/giWvlN3). Os números no gráfico representam a quantidade de vagas que pagam naquela faixa salarial.

Se você gosta mais da parte de programação, creio que o backend vai te oferecer muito mais desafios, em quantidade e profundidade/dificuldade, então vai ser uma área que a longo prazo vai te trazer mais satisfação.

Para quem gosta mais da parte de UI, obviamente o frontend é o mais indicado, mas acho bem importante que o profissional se aprofunde também nos conceitos de design, pois no futuro (próximo?!) a parte que conhecemos de "recorte de layout" tende a ser automatizada, então o importante nisso será o conceito/habilidade na criação da interface, pois a tradução para código será feita por algum programa.

É uma questão profunda essa que você perguntou, e a resposta irá sair de você mesmo. Só conseguiremos te mostrar alguns dados e com base nisso você irá se decidir.

Um ponto importante a se considerar é que, comprovadamente, dinheiro não traz felicidade. Veja mais sobre isso [aqui](https://www.ted.com/talks/mihaly_csikszentmihalyi_on_flow). Sendo assim, mesmo que tenha mais vagas em uma área ou outra, ou elas paguem mais, isso não importa no fim. O que importa é o que você gosta de fazer.

Uma pergunta que você pode se fazer: quando estou trabalhando, qual parte me dá mais prazer em fazer? Frontend, Backend ou gosto igual? Se gostar igual, você pode seguir pelo caminho full-stack.

Outra questão que vejo é o ponto de você dizer Front vs PHP.

Cara, tem muita tecnologia interessante no backend, então se você optar por esse caminho, você tem que estar ciente que ele vai bem além do PHP, e acho que em geral as vagas para PHP pagam menos que outras tecnologias, então como você citou a questão salarial, procuraria ver quais tecnologias oferecem melhores oportunidades nesse quesito.

**Comment by ericdouglas** [comment]
@bernardodiasc excelentes perguntas propostas no seu texto, temos que sempre nos reavaliar com base nesse tipo de questões para irmos ajustando o norte. 

Lembrando que a cada época da vida as respostas podem mudar, e ai temos que realinhar as coisas.

Bem conversa de divã mesmo :smile:

**Comment by bernardodiasc** [comment]
Uma das experiências mais enriquecedoras que vivi em termos de auto-avaliação foi no Empretec (http://www.sebrae.com.br/sites/PortalSebrae/sebraeaz/empretec-fortaleca-suas-habilidades-como-empreendedor,db3c36627a963410VgnVCM1000003b74010aRCRD, http://www.sebrae.com.br/sites/PortalSebrae/ufs/sc/noticias/por-que-voce-deve-fazer-o-empretec,2853a18de44a8410VgnVCM1000003b74010aRCRD).

O empreendedorismo não é uma habilidade somente para empresários, é algo que qualquer profissional pode desenvolver, é uma habilidade ainda mais pessoal do que profissional.

Pra quem não curte viver a vida cumprindo ordem e seguindo comando dos outros, pra quem quer ser líder de si mesmo, recomendo. :)

**Comment by matheusggds** [comment]
Obrigado galera, desculpa o tópico sair da idéia do fórum. Levarei em contato tudo que cada um disse, me ajudaram muito, obrigado novamente.

**Comment by gabrieldarezzo** [comment]
(Relatos de um programador php)

Estava no mesmo dilema ano passado....
Na verdade muito parecido com o seu caso, estava em agencia digital fazendo vários trampos pequenos e o máximo que eu explorava em php era um plugin aqui, uma adaptação ali outro CRUD e ficava por isso mesmo.

No final do ano passado acabou aparecendo uma oportunidade de trabalhar numa empresa focada em desenvolvimento de sistema educacional.

Está sendo legal e talz, porem uma coisa que me incomoda muito é a qualidade de código sabe? 
Ultimamente tenho estudado muito qualidade de código (a noite em casa sem prazo, sem chefe) 
E aplicando oq é possível no ambiente real/hostil.

Key Words:
TDD, Design Patterns, OO, SOLID, PSR, Full-Frameworks.

--------

(Relatos de um aprendiz em JS)
Nessa mesma empresa apareceu a oportunidade de fazer um WebApp/PWA/App hibrido, oq está sendo legal para estudar tudo relacionado a esse mundo fantástico de Javascript, abandonar bastante o jQuery e focar novamente em qualidade do código porem em JS.

Fica legal essa intercalação pois conheço ponta a ponta desde criação do WireFrame/validação da funcionalidade com o cliente até a implementação de um cron no servidor linux....

Key Words:
TDD, Design Patterns, OO, SOLID, [PREENCHA SEU FW-JS] 

-------
Sim as key words se repetem independente da 'armadura' que estiver no dia....


Resumindo, independente do local de trabalho a busca por melhoria deve ser constante mesmo se o ambiente que você está não permita



> Cara, tem muita tecnologia interessante no backend, então se você optar por esse caminho, você tem que estar ciente que ele vai bem além do PHP, e acho que em geral as vagas para PHP pagam menos que outras tecnologias, então como você citou a questão salarial, procuraria ver quais tecnologias oferecem melhores oportunidades nesse quesito.


@ericdouglas Concordo em gênero, grau e especie.


Links legais:
https://github.com/frontendbr/forum#perguntas-frequentes
http://hipsters.tech/carreira-em-tecnologia-hipsters-31/
http://hipsters.tech/php-nao-e-coisa-de-sobrinho-hipsters-36/
https://www.alura.com.br/

**Comment by wellingtonpgp** [comment]
Domine a técnica, não apenas algumas tecnologias. 😀
